### ISSUE TITLE
Error with 2.0.17 and lvs_sync on the backup

### ISSUE THREAD

**Issue by aholtzmbw** [acassen/keepalived#1337]
We are running 2.0.17 on a pair of Ubuntu 18.04 servers using the following kernel:
5.0.0-19-generic #20~18.04.1-Ubuntu SMP Thu Jun 20 12:17:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

Our lvs_sync line is pretty basic:
lvs_sync_daemon ens192 VI_external

When the master keepalived server is up and running the following errors are seen in the backup server logs when you try and bring keepalived online:
IPVS cmd IP_VS_SO_SET_STOPDAEMON(1164) error: Interrupted system call(4)
or
IPVS cmd IP_VS_SO_SET_STOPDAEMON(1164) error: No such file or directory(2)

When running "ipvsadm -L --daemon" on the backup there is no output.  The master server does have a master sync process running as expected.

If the master keepalived is shutdown then a master sync process is created on the backup server as evidenced from the output of ipvsadm.  If the master comes back online then the sync process on the backup is terminated and no backup sync process exists (again verified via ipvsadm).

However, if the master and backup are both shut down and you only bring up the backup then it will create a master sync process (as expected).  If you then start up the master the master sync process on the backup server will transition to a backup sync process and work as expected.

So in short the bug I'm seeing is related to bringing up the backup and having it create a backup sync process from the get-go. If it comes up on its own then it will transition that master sync process to a backup properly.

We do not see this behavior under 2.0.16 on these same servers (we've flipped the binaries out without changing anything in keepalived.conf).

**Comment by aholtzmbw** [comment]
I can confirm this problem also exists on CentOS 7.5 with this kernel info:
**3.10.0-862.3.3.el7.x86_64 #1 SMP Fri Jun 15 04:15:27 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux**

Similar log errors when starting up the backup:

> kernel: IPVS: __ip_vs_del_service: enter
> kernel: IPVS: stopping backup sync thread 1008 ...
> Keepalived_vrrp[55274]: IPVS cmd IP_VS_SO_SET_STOPDAEMON(1164) error: No such file or directory(2)
> kernel: IPVS: stopping backup sync thread 55275 ...
>

**Comment by pqarmitage** [comment]
The problem is commit eb929f8. I am working on a path for it now.

**Comment by pqarmitage** [comment]
Commit e4d97ee resolves this issue. Commit eb929f8 moved the shutdown of the sync daemon to the wrong place - it fixed the reported problem, but unfortunately when testing the fix I didn't notice that it caused another problem.

**Comment by aholtzmbw** [comment]
Awesome - thanks.  I'll pull/install that patch and let you know how it goes.

**Comment by aholtzmbw** [comment]
Everything works as I'd expect using this commit.
